pgcli/main.py

```diff
@@ -423,6 +349,133 @@ def prompt_tokens(cli):
             logger.debug('Restoring env var LESS to %r.', original_less_opts)
             os.environ['LESS'] = original_less_opts
 
+    def _build_cli(self):
+
+        def set_vi_mode(value):
+            self.vi_mode = value
+
+        key_binding_manager = pgcli_bindings(
+            get_vi_mode_enabled=lambda: self.vi_mode,
+            set_vi_mode_enabled=set_vi_mode)
+
+        def prompt_tokens(_):
+            return [(Token.Prompt, '%s> ' % self.pgexecute.dbname)]
+
+        get_toolbar_tokens = create_toolbar_tokens_func(
+            lambda: self.vi_mode, self.completion_refresher.is_refreshing)
+
+        layout = create_default_layout(
+            lexer=PostgresLexer,
+            reserve_space_for_menu=True,
+            get_prompt_tokens=prompt_tokens,
+            get_bottom_toolbar_tokens=get_toolbar_tokens,
+            display_completions_in_columns=self.wider_completion_menu,
+            multiline=True,
+            extra_input_processors=[
+               # Highlight matching brackets while editing.
+               ConditionalProcessor(
+                   processor=HighlightMatchingBracketProcessor(chars='[](){}'),
+                   filter=HasFocus(DEFAULT_BUFFER) & ~IsDone()),
+            ])
+
+        history_file = self.config['main']['history_file']
+        with self._completer_lock:
+            buf = PGBuffer(
+                always_multiline=self.multi_line,
+                completer=self.completer,
+                history=FileHistory(os.path.expanduser(history_file)),
+                complete_while_typing=Always())
+
+            application = Application(
+                style=style_factory(self.syntax_style, self.cli_style),
+                layout=layout,
+                buffer=buf,
+                key_bindings_registry=key_binding_manager.registry,
+                on_exit=AbortAction.RAISE_EXCEPTION,
+                ignore_case=True)
+
+            cli = CommandLineInterface(
+                application=application,
+                eventloop=create_eventloop())
+
+            return cli
+
+    def _evaluate_command(self, text):
+        """Used to run a command entered by the user during CLI operation
+        (Puts the E in REPL)
+
+        returns (results, MetaQuery)
+        """
+        logger = self.logger
+        logger.debug('sql: %r', text)
+
+        all_success = True
+        meta_changed = False  # CREATE, ALTER, DROP, etc
+        mutated = False  # INSERT, DELETE, etc
+        db_changed = False
+        path_changed = False
+        output = []
+        total = 0
+
+        # Run the query.
+        start = time()
+        on_error_resume = self.on_error == 'RESUME'
+        res = self.pgexecute.run(text, self.pgspecial,
+                                 exception_formatter, on_error_resume)
```

Comment:
I noticed that pgexecute is not updated, it still relies on the presence of `exception_formatter` to decide whether to raise an exception or not. 

Which means `pgexecute` will never raise an exception because we're always setting the `exception_formatter`.